net-utils/src/ip_echo_server.rs

```diff
@@ -23,6 +27,14 @@ pub(crate) struct IpEchoServerMessage {
     udp_ports: [u16; MAX_PORT_COUNT_PER_MESSAGE], // Fixed size list of ports to avoid vec serde
 }
 
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct IpEchoServerResponse {
+    // Public IP address of request echoed back to the node.
+    pub(crate) address: IpAddr,
+    // Cluster shred-version of the node running the server.
+    pub(crate) shred_version: Option<u16>,
```

Comment:
Oh I think we might be able to use `#[serde(deserialize_with = "default_on_eof")]` here so that the new `IpEchoServerResponse` retains compatibility with the current endpoints that will only return  `address: IpAddr`

Reply:
added, thanks